src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +14,29 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            session.beginTransaction();
+            session.persist(country);
+            session.getTransaction().commit();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't add country: " + country, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
```

Comment:
Consider using try-with-resources for the `Session` in the `add` method to ensure it is closed properly, even if an exception occurs. This will make the code cleaner and more robust.